client/src/pages/Map.js

```diff
@@ -115,27 +103,22 @@ export default function Map(
                     <p>Address: {selectedRestaurant.street}, {selectedRestaurant.postal_code} {selectedRestaurant.city}</p>
                     <p>Menu: {selectedRestaurant.website}</p>
                     <p>Phone: {selectedRestaurant.phone}</p>
-                    <button onClick={() => addFavoriteRestaurant(selectedRestaurant)}>
-                      <Heart />
-                    </button>
+                    <FavoriteButton                 
+                      isFavorite={favoriteRestaurants.some(
+                        (favoriteRestaurant) => selectedRestaurant._id === favoriteRestaurant._id
+                      )} 
+                      onClick={() => addFavoriteRestaurant(selectedRestaurant)}>
```

Comment:
I might have seen this or a similar function (`isFavorite()`) before that could be reused :)